src/transformState.js

```diff
@@ -5,7 +5,33 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  let changedState = state;
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        changedState = Object.assign(changedState, action.extraData);
+        break;
+      }
+
+      case 'removeProperties': {
+        for (const key of action.keysToRemove) {
+          delete changedState[key];
+        }
+        break;
+      }
+
+      case 'clear': {
+        for (const property in changedState) {
+          delete changedState[property];
+        }
+        break;
+      }
+
+      default:
+        throw new Error('Unknown action');
```

Comment:
Not critical, but can we add also a type of action in error, so it will be more descriptive?